### PR TITLE
[10.0] Migration of account_asset_management

### PR DESCRIPTION
**Pull request by adrienpeiffer**
Based on the work of @florian-dacosta in https://github.com/akretion/account-financial-tools/tree/9-account_asset_management

- [x] Test migration script

### CODE REVIEW

**Comment by adrienpeiffer** [comment]
@florian-dacosta I am wondering about some modification.
- Why have you changed ```asset.asset.category``` model name
- Why have you changed ```asset_value``` field name

Can you clarify that ?

**Comment by florian-dacosta** [comment]
For the model names, Luc proposed to change it so it would be different from odoo native module.
So there is not conflict between table... (even if those 2 modules should not be installed together)

For the field asset_value, we agreed with Luc that the name was confusing so we changed it in a name more meaningful.

**Comment by yvaucher** [comment]
Still WIP ? (changing tag according to title)

**Comment by adrienpeiffer** [comment]
@yvaucher No :)

**Comment by luc-demeyer** [comment]
I am currently working on  a version  with full ORM compliancy (no cr.execute any more), removal of stored computed fields on parent_ids and a lot of bug fixes.
Initial testing shows enormous performance improvements (I need to improve performance for a project with 18.000 financial assets). I hope to have a first PR ready by the end of this month.
Seen the number of bugs I found in the current code, this PR is not ready yet for merge.

**Comment by luc-demeyer** [comment]
Have a look to https://github.com/OCA/account-financial-tools/pull/501.
The V80 code is now fully aligned to the V10 db structure with as only difference the 'date_range' object.
PR 501 also resolves the travis conflict between the official addons 'account_asset' module and the 'account_asset_asset' module. Hence PR 501 is completely green considered from the 'asset management' standpoint (still some greenify work to on a couple of unrelated errors, e.g. account_journal_period_close).
The work in PR 501 is largely based upon the V9 and V10 porting work.

I intend to work on the forward port of PR 501 to v10 next weekend, hence we are nearly there.

**Comment by gurneyalex** [comment]
@adrienpeiffer can you check https://github.com/acsone/account-financial-tools/pull/5 ?

**Comment by gurneyalex** [comment]
@adrienpeiffer I tried this module and I have a circular dependency issue with the parent asset : this field is mandatory and I cannot create a new asset manually because I don't have a parent asset. 

I have a similar bootstrapping issue with the asset category: the field is mandatory when I create an asset, but inside the category creation form, I have a mandatory asset_id field 

=> I'm stuck and cannot create a new asset category or a new asset. 

Am I missing something obvious? cc @fclementic2c

**Comment by adrienpeiffer** [comment]
Hi @gurneyalex,

First, I'm glad to see you test this module :smile:

To create a parent asset, you should change the type of the asset to view.  So, with this configuration, the parent field is no longer required. After you should able to create an asset category by using this view asset and so, to create an asset with this category.

**Comment by gurneyalex** [comment]
Oh, yes... Thanks :-)

**Comment by gurneyalex** [comment]
@adrienpeiffer preliminary tests look good thanks for this hard work. 

Do you happen to know what the status of account_asset_management_xls is? report_xls is gone, and this report used to rely on a lot of features from report_xls not available in report_xlsx...

File: account_asset_management/models/account_asset_line.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                        _("You cannot set the date on a depreciation line "
+                          "prior to already posted entries."))
+        return super(AccountAssetLine, self).write(vals)
+
+    @api.multi
+    def _setup_move_data(self, depreciation_date):
+        self.ensure_one()
+        asset = self.asset_id
+        move_data = {
+            'name': asset.name,
```
Comment (gurneyalex) [inline account_asset_management/models/account_asset_line.py:195]: it is really weird not to have the next value from the sequence of the journal here. 

cc @fclementic2c

**Comment by fclementic2c** [comment]
I did quick functional test and I noticed that:
* You cannot compute depreciation board without the fiscal year date range created in the future... We should not have such a constraint.
nodaterange

nofy_undifined

* If you change the date_start (on the asset) : the value_depreciated field does not recompute / reinitilize
screenshot-3316206-441-15b9bb runbot2 odoo-community org-2018-01-31-13-43-32-561

**Comment by adrienpeiffer** [comment]
I close this one as #677  is merged.

**Pull request closed without merging**